netboot-services/ipxeMenuGenerator/main.go

```diff
@@ -0,0 +1,198 @@
+	if err != nil {
+		fmt.Println("Error:", err)
+		panic(err)
+	}
+
+	var matches []fs.DirEntry
+	for _, file := range files {
+		if strings.HasSuffix(file.Name(), ".squashfs") {
+			matches = append(matches, file)
+		}
+	}
+
+	sort.Sort(ByModTime(matches))
+
+	if len(matches) > 0 {
+		return strings.TrimSuffix(matches[0].Name(), ".squashfs")
+	}
+	panic("No squashfs image found")
+}
+
+type kernelVersion struct {
+	KernelVersion string `json:"version"`
+}
+
+func getMatchingKernelVersion(folderName string, imageName string) string {
+	var version kernelVersion
+
+	if strings.HasPrefix(folderName, ".azDownload") {
+		return ""
+	}
+
+	bytes, err := os.ReadFile(fmt.Sprintf("%s/%s-kernel.json", folderName, imageName))
+	if err != nil {
+		panic(err)
+	}
+	err = json.Unmarshal(bytes, &version)
+	if err != nil {
+		panic(err)
+	}
+	return version.KernelVersion
+}
+
+func renderMenuIpxe(filename string, folderName string, netbootServerIP string) {
+	mostRecentSquashfsImageName := getMostRecentSquashfsImage(folderName)
+	kernelVersionString := getMatchingKernelVersion(folderName, mostRecentSquashfsImageName)
+	if kernelVersionString == "" {
+		return
+	}
+
+	j2, err := jinja2.NewJinja2("menu.ipxe", 1,
+		jinja2.WithGlobal("netbootServerIP", netbootServerIP),
+		jinja2.WithGlobal("imageName", mostRecentSquashfsImageName),
+		jinja2.WithGlobal("kernelFolderName", kernelVersionString),
+	)
+	if err != nil {
+		panic(err)
+	}
+	defer j2.Close()
+
+	renderedString, err := j2.RenderFile(filename)
+	if err != nil {
+		panic(err)
+	}
+
+	filePath := fmt.Sprintf("/menus/%s", strings.ReplaceAll(filename, ".j2", ""))
+	os.WriteFile(filePath, []byte(renderedString), 0644)
+	fmt.Printf("filename: %s\nresult: %s", filename, renderedString)
+}
+
+func getDevImages() []image {
+	return getImages(DevFolder)
+}
+
+func getProdImages() []image {
+	return getImages(ProdFolder)
+}
+
+func getImages(folderName string) []image {
+	var images []image
+
+	files, err := os.ReadDir(folderName)
+	if err != nil {
+		fmt.Println("Error:", err)
+		panic(err)
+	}
+
+	var matches []fs.DirEntry
+	for _, file := range files {
+		if strings.HasSuffix(file.Name(), ".squashfs") {
+			if strings.HasPrefix(file.Name(), ".azDownload") {
+				continue
+			}
+			matches = append(matches, file)
+		}
+	}
+
+	sort.Sort(ByModTime(matches))
+
+	for _, file := range matches {
+		imageName := strings.TrimSuffix(file.Name(), ".squashfs")
+		kernelVersionString := getMatchingKernelVersion(folderName, imageName)
+		images = append(images, image{ImageName: imageName, KernelVersion: kernelVersionString})
+	}
+
+	return images
+}
+
+func renderAdvancedMenu(filename string, netbootServerIP string) {
+	prodImages := getProdImages()
+	devImages := getDevImages()
+	j2, err := jinja2.NewJinja2("advancedmenu.ipxe", 1,
+		jinja2.WithGlobal("netbootServerIP", netbootServerIP),
+		jinja2.WithGlobal("prod", prodImages),
+		jinja2.WithGlobal("dev", devImages),
+	)
+	if err != nil {
+		panic(err)
```

Comment:
Is this service still functional, if we can't render the advanced menu? I would argue that it is somewhat functional, so we shouldn't panic here